src/ast/element-descriptor.ts

```diff
@@ -29,3 +43,33 @@ export interface ElementDescriptor extends Descriptor {
 
   abstract?: boolean;
 }
+
+/**
+ * The metadata for a single polymer element
+ */
+export class ElementDescriptor implements Descriptor {
```

Comment:
Yeah that makes sense. For that we'll likely  want to move some of the code I've been writing as part of the serializer up into the element finder, to convert the polymer-specific stuff into general custom element metadata.